src/protocol/ipa/mod.rs

```diff
@@ -522,7 +522,7 @@ pub mod tests {
             (Fp31, MatchKey, BreakdownKey)
         );
 
-        let result: Vec<GenericReportTestInput<Fp31, MatchKey, BreakdownKey>> = world
```

Comment:
This seems like a regression. Why not call `reconstruct`?

Reply:
Unfortunately reconstruct doesn't exist for malicious. Another option is to return a validator, validate results and then check. I was not sure if we want to return validator at the end of ipa